src/event-handlers/impl/ProjectLicenseEventHandler.js

```diff
@@ -0,0 +1,91 @@
+import BaseEventHandler from './../base/BaseEventHandler';
+import APP_EVENT from './../../events/base/AppEvent';
+import { ProjectLicensingService, ProjectLicensingDtoService } from './../../services';
+import { DEFAULT_STATUS } from './../../constants/';
+import { CONTRACT_AGREEMENT_TYPE } from '@deip/constants';
+
+class ProjectLicenseEventHandler extends BaseEventHandler {
+  constructor() {
+    super();
+  }
+}
+
+const projectLicenseEventHandler = new ProjectLicenseEventHandler();
+
+const projectLicensingService = new ProjectLicensingService();
+const projectLicensingDtoService = new ProjectLicensingDtoService();
+
+projectLicenseEventHandler.register(APP_EVENT.CONTRACT_AGREEMENT_PROPOSAL_CREATED, async (event) => {
+
+  const {
+    contractAgreementId,
+    proposalId,
+    creator,
+    terms,
+    type,
+  } = event.getEventPayload();
+
+  if (type === CONTRACT_AGREEMENT_TYPE.PROJECT_LICENSE) {
+    await projectLicensingService.createProjectLicense({
+      licenseId: contractAgreementId,
+      owner: creator,
+      requestId: proposalId,
+      projectId: terms.projectId,
+      licensePlan: terms,
+      status: DEFAULT_STATUS.PENDING
+    });
+  }
+});
+
+projectLicenseEventHandler.register(APP_EVENT.CONTRACT_AGREEMENT_PROPOSAL_DECLINED, async (event) => {
+
+  const { contractAgreementId, type } = event.getEventPayload();
+
+  if (type === CONTRACT_AGREEMENT_TYPE.PROJECT_LICENSE) {
+    await projectLicensingService.updateProjectLicense({
+      _id: contractAgreementId,
+      status: DEFAULT_STATUS.REJECTED
+    });
+  }
+});
+
+projectLicenseEventHandler.register(APP_EVENT.CONTRACT_AGREEMENT_CREATED, async (event) => {
+
+  const {
+    entityId: contractAgreementId,
+    creator,
+    terms,
+    type,
+  } = event.getEventPayload();
+
+  if (type === CONTRACT_AGREEMENT_TYPE.PROJECT_LICENSE) {
+    const license = await projectLicensingDtoService.getProjectLicense(contractAgreementId);
+    if (!license) {
+      await projectLicensingService.createProjectLicense({
+        licenseId: contractAgreementId,
+        owner: creator,
+        projectId: terms.projectId,
+        licensePlan: terms,
+        status: DEFAULT_STATUS.PENDING
+      });
+    }
+  }
+});
+
+projectLicenseEventHandler.register(APP_EVENT.CONTRACT_AGREEMENT_ACCEPTED, async (event) => {
+
+  const { entityId: contractAgreementId, party } = event.getEventPayload();
+
+  const license = await projectLicensingDtoService.getProjectLicense(contractAgreementId);
+  if (license) {
+    const isAllAccepted = license.chainContract.parties.every((p) => p[1] === 'Accepted')
```

Comment:
Add additional array field to ContractAgreemenSchema called accepteByPartiesand populate this field each time on CONTRACT_AGREEMENT_ACCEPTED event. Define the status by comparing parties and acceptedByParties arrays